src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -60,36 +73,76 @@ private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progre
                 case ProgressStatus.Started:
                     StartedOrStopped(started: true);
                     break;
-                case ProgressStatus.Updated:
-                    ProgressUpdated(progressData.FilePathOpt);
+                case ProgressStatus.PendingItemUpdated:
+                    _lastPendingItemCount = progressData.PendingItemCount.Value;
+                    ProgressUpdated();
                     break;
                 case ProgressStatus.Stoped:
                     StartedOrStopped(started: false);
                     break;
+                case ProgressStatus.Evaluating:
+                case ProgressStatus.Paused:
+                    _lastProgressStatus = progressData.Status;
+                    ProgressUpdated();
+                    break;
                 default:
                     throw ExceptionUtilities.UnexpectedValue(progressData.Status);
             }
         }
 
-        private void ProgressUpdated(string filePathOpt)
+        private void ProgressUpdated()
         {
+            // we prefer showing evaluating if progress is flipping between evaluate and pause
+            // in short period of time.
+            var forceUpdate = _lastShownProgressStatus == ProgressStatus.Paused &&
+                              _lastProgressStatus == ProgressStatus.Evaluating;
+
             var current = DateTimeOffset.UtcNow;
-            if (current - _lastTimeReported < s_minimumInterval)
+            if (!forceUpdate && current - _lastTimeReported < s_minimumInterval)
```

Comment:
❓ Why are we doing an interval check instead of checking if `_resettableDelay` is null?

Reply:
_resettableDelay makes sure that we at the end update status to correct value. _lastTimeReported make sure that we only update at least s_minimumInterval interval.

when an event stream comes in as below (assuming minimum interval is 200ms)

e1 -> (100ms)-> e2 -> (300ms)-> e3 -> (100ms) -> e4

actual status shown to users will be without _resettableDelay

e1 -> e3.

e2 and e4 will be skipped since internal is smaller than min interval. 

losing e2 is fine, but e4 is problematic since the user now sees the wrong status until the next event comes in. for example, it shows "Evaluating" when it is actually "Paused" until the next event.

what _resettableDelay does is making sure that if the next event doesn't come in within 200ms, it updates status to e4.